include/kf/stl/map

```diff
@@ -81,6 +121,159 @@ namespace kf
             return m_internalMap->emplace(std::forward<Args>(values)...);
         }
 
+        std::optional<std::pair<iterator, bool>> insert(const value_type& value)
+        {
+            if (!m_internalMap->get_allocator().prepareMemory(kNodeSize))
+            {
+                return {};
+            }
+
+            return m_internalMap->insert(value);
+        }
+
+        std::optional<std::pair<iterator, bool>> insert(value_type&& value)
+        {
+            if (!m_internalMap->get_allocator().prepareMemory(kNodeSize))
+            {
+                return {};
+            }
+
+            return m_internalMap->insert(std::move(value));
+        }
+
+        std::optional<iterator> insert(const_iterator hint, const value_type& value)
+        {
+            if (!m_internalMap->get_allocator().prepareMemory(kNodeSize))
+            {
+                return {};
+            }
+
+            return m_internalMap->insert(hint, value);
+        }
+
+        std::optional<iterator> insert(const_iterator hint, value_type&& value)
+        {
+            if (!m_internalMap->get_allocator().prepareMemory(kNodeSize))
+            {
+                return {};
+            }
+
+            return m_internalMap->insert(hint, std::move(value));
+        }
+
+        template<class InputIt>
+        std::optional<bool> insert(InputIt first, InputIt last)
+        {
+            // For range insert, we prepare memory for each element as we go
+            // This is a conservative approach that matches the allocator design
+            for (auto it = first; it != last; ++it)
+            {
+                if (m_internalMap->find(it->first) == m_internalMap->end())
+                {
+                    if (!m_internalMap->get_allocator().prepareMemory(kNodeSize))
```

Comment:
`prepareMemory` can't be called multiple times in a row to allocate memory for multiple elements. It should be used this way: allocate, consume, allocate, consume.